tests/test_align2d.py

```diff
@@ -0,0 +1,223 @@
+            self.basis, n_angles=self.n_search_angles
+        ).align(self.classes, self.reflections, self.coefs)
+
+        self.assertTrue(np.all(_classes == self.classes))
+        self.assertTrue(np.all(_reflections == self.reflections))
+        self.assertIsNone(_shifts)
+
+        # Crude check that we are closer to known angle than the next rotation
+        self.assertTrue(np.all((_rotations - self.thetas) <= (self.step / 2)))
+
+        # Fine check that we are within n_angles.
+        self.assertTrue(
+            np.all((_rotations - self.thetas) <= (2 * np.pi / self.n_search_angles))
+        )
+
+
+class BFSRAlign2DTestCase(BFRAlign2DTestCase):
+
+    aligner = BFSRAlign2D
+
+    def setUp(self):
+        # Inherit basic params from the base class
+        super(BFRAlign2DTestCase, self).setUp()
+
+        # Setup shifts, don't shift the base image
+        self.shifts = np.zeros((self.n_img, 2))
+        self.shifts[1:, 0] = 2
+        self.shifts[1:, 1] = 4
+
+        # Execute the remaining setup from BFRAlign2DTestCase
+        super().setUp()
+
+    def testNoShift(self):
+        """
+        Test we raise an error when our basis does not provide `shift` method.
+        """
+
+        # DiracBasis does not provide `rotate` or `shift`.
+        basis = DiracBasis((self.resolution, self.resolution), dtype=self.dtype)
+
+        # The missing `rotate` case was already covered by (inherited) NoRot.
+        # Add a dummy rotate method; we will still be missing `shift`,
+        basis.rotate = lambda x: x
+
+        # and that should raise an error during instantiation.
+        with pytest.raises(RuntimeError, match=r".* must provide a `shift` method."):
+            _ = self.aligner(basis)
+
+    def testAlign(self):
+        """
+        Construct a stack of images with known rotations.
+
+        Rotationally align the stack and compare output with known rotations.
+        """
+
+        # Construction the Aligner and then call the main `align` method
+        _classes, _reflections, _rotations, _shifts, _ = self.aligner(
+            self.basis, n_angles=self.n_search_angles, n_x_shifts=1, n_y_shifts=1
+        ).align(self.classes, self.reflections, self.coefs)
+
+        self.assertTrue(np.all(_classes == self.classes))
+        self.assertTrue(np.all(_reflections == self.reflections))
+
+        # Crude check that we are closer to known angle than the next rotation
+        self.assertTrue(np.all((_rotations - self.thetas) <= (self.step / 2)))
+
+        # Fine check that we are within n_angles.
+        self.assertTrue(
+            np.all((_rotations - self.thetas) <= (2 * np.pi / self.n_search_angles))
+        )
+
+        # Check that we are _not_ shifting the base image
+        self.assertTrue(np.all(_shifts[0][0] == 0))
+        # Check that we produced  estimated shifts away from origin
+        self.assertTrue(np.all(np.hypot(*_shifts[0][1:].T) >= 1))
```

Comment:
shifting then rotating (what I called BFSR) isn't the same as rotating then shifting( simulation source).

Here I'm just sanity checking that the code is observing improvement by moving away from the origin. The [2,4] is to just ensure we move enough. I was hoping to avoid getting really deep into the roto-translational checking. You bring up a good point, maybe we could make this part of a tutorial later when we develop more mature alignment codes.  There we can draw such things.